### PR TITLE
Fixes #589: Text-to-Speech feature

### LINKED ISSUE #589: Improve speech to text on search interface
Activating voice results in a continuous activated microphone. Please test the feature on the market leader page and adapt Susper to work in a similar way. First Google shows a screen with "Speak Now" and outputs the first info in voice when there is a sidebar box, e.g. "According to Wikipedia...".

### PR DESCRIPTION
**Pull request by harshit98**
Fixes issue #589 

Changes: 
- Adds text-to-speech feature.

Demo Link: https://harshit98.github.io/susper.com/

Screenshots for the change: 

(Not applicable here)

Kindly review: @nikhilrayaprolu @Marauderer97, whenever the `infobox` component is being loaded, only 'undefined' is uttered. How to proceed so, that it reads the description in the infobox?

### CODE REVIEW

File: src/app/infobox/infobox.component.ts [hunk 1]
```diff
@@ -41,11 +48,10 @@ export class InfoboxComponent implements OnInit {
         }
 
     });
-
-
   }
 
   ngOnInit() {
+    this.synthesis.speak(this.results[0]);
```
Comment (nikhilrayaprolu) [inline src/app/infobox/infobox.component.ts]: we won't receive results immediately when a component is initialised. you have to call this function only after we receive results from the server.

Comment (harshit98) [inline src/app/infobox/infobox.component.ts]: @nikhilrayaprolu I'm working on it.

**Comment by harshit98** [comment]
@nikhilrayaprolu Done as suggested. Now nothing is being uttered.

File: src/app/infobox/infobox.component.ts [hunk 2]
```diff
@@ -41,8 +49,14 @@ export class InfoboxComponent implements OnInit {
         }
 
     });
+  }
 
+  if (synthesisEnable = true) {
```
Comment (nikhilrayaprolu) [inline src/app/infobox/infobox.component.ts]: @harshit98 while you are coding something in javascript, remember that it is an asynchronous language.
where is the correct block in the whole code which runs immediately after receiving results?
it the block between 39 to 51 that is the block inside `this.response$.subscribe`

Comment (harshit98) [inline src/app/infobox/infobox.component.ts]: Done @nikhilrayaprolu. I tried to pass `this.results` the value in the function. But it utters 'object object'

File: src/app/infobox/infobox.component.ts [hunk 3]
```diff
@@ -32,6 +40,7 @@ export class InfoboxComponent implements OnInit {
         if (res.results[0]) {
           if (res.results[0].label.toLowerCase().includes(this.keyword.toLowerCase())) {
             this.results = res.results;
+            this.startSpeaking(this.results);
```
Comment (nikhilrayaprolu) [inline src/app/infobox/infobox.component.ts]: good work pass `this.results[0].description`

**Comment by harshit98** [comment]
@mariobehling @nikhilrayaprolu @Marauderer97 The text-to-speech feature is now working well. Kindly review.

**Comment by nikhilrayaprolu** [comment]
@harshit98 the link isn't working for me.

**Comment by harshit98** [comment]
@nikhilrayaprolu It is working fine. Try to run on Firefox.

**Comment by harshit98** [comment]
@nikhilrayaprolu 
- Test link is working on Chrome now as well. There might be some problem with GitHub pages. Test link: https://susper-pr-673.herokuapp.com/
- It will require discussion to activate this service using voice search because it is being activated when info box component loads. 
- Now the info box component loads up if: 1) query is entered manually. 2) using voice search. So right now, the TTS feature will work in both ways. It would be better to open up a separate issue for it.

**Comment by nikhilrayaprolu** [comment]
@harshit98 it is still not working for me on chrome for the new link

- voice should be activated only if user input is through voice, and only when info box is loaded. use these two conditionals.

**Comment by Marauderer97** [comment]
@harshit98 Can you provide a small gif of how the feature is working for you on chrome on your link since it is not working for me

**Comment by harshit98** [comment]
@nikhilrayaprolu Done! how should I call TTS service? Rest of the work has been done as discussed on susper chat channel.

**Comment by nikhilrayaprolu** [comment]
@harshit98 please look at this commit and follow similarly to implement the conditional https://github.com/nikhilrayaprolu/susper.com-1/commit/d6264dfa0c16a881dd7f7ab772e59430187062a4

**Comment by harshit98** [comment]
@nikhilrayaprolu @Marauderer97 Working as expected. Please review. :)

**Comment by nikhilrayaprolu** [comment]
@harshit98 I am not able to receive any results on chrome, some errors pop up in console please check.

**Comment by harshit98** [comment]
@Marauderer97 @nikhilrayaprolu If demo link is not working, here : <a href="https://www.youtube.com/watch?v=pkw4_co7PaU&feature=youtu.be">Video link</a>

**Comment by nikhilrayaprolu** [comment]
@harshit98 please look at this changes. 
1.the voice is not working on the index page, it is working only on the results page.
2.the voice is not stopping once it has completely read the information, it is re reading it in an infinite loop.

**Comment by harshit98** [comment]
Done @nikhilrayaprolu :+1:

**Comment by mariobehling** [comment]
Still not working for me.

**Pull request merged**